### PR TITLE
Anima update

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
-# encoding: utf-8
-
 source 'https://rubygems.org'
 
 gemspec name: 'mutant'
 
 gem 'devtools', git: 'https://github.com/rom-rb/devtools.git'
+eval_gemfile File.expand_path('../Gemfile.shared', __FILE__)
```
Comment (dkubb) [inline Gemfile]: The `Gemfile.shared` file is empty. What is the reason for evaling it? Why not just remove the file and this line.

File: lib/mutant/cli.rb [hunk 2]
```diff
@@ -207,7 +207,7 @@ def update(attributes)
     #
     # @api private
     def add(attribute, value)
-      update(attribute => config.public_send(attribute).dup << value)
+      with(attribute => config.public_send(attribute).dup << value)
```
Comment (dkubb) [inline lib/mutant/cli.rb]: I know in the past you've preferred to use different constructs for this, like:

```ruby
config.public_send(attribute) + [value]
```

File: lib/mutant/matcher/config.rb [hunk 3]
```diff
@@ -41,7 +41,7 @@ def inspect
       #
       # @api private
       def add(attribute, value)
-        update(attribute => public_send(attribute).dup << value)
+        with(attribute => public_send(attribute).dup << value)
```
Comment (dkubb) [inline lib/mutant/matcher/config.rb]: I have the same comment for this as https://github.com/mbj/mutant/pull/439/files#r38951709

File: Gemfile (hunk 1 above)
Comment (mbj) [inline Gemfile]: There should be a commend in that file explaining the reason why it exist. If its not there anymore I rebased it away by accident.

Always when I update mutants dependencies (especially morpher and anima) there is a state where dependencies need to be pulled from git sources.

These git sources need to be reflected in the Gemfiles for:

* The vendored test app, per rspec (and future minitest) integration
* The corpus tests Gemfiles

That would be ~5 Gemfiles (tendency to grow) I'd need to manually add (remove) the git source from/to.

Now instead: I can simply use this (now empty) file to ease upgrades of shared dependencies.

As I said: there was a note in the `Gemfile.shared` trying to explain its presence as (now empty) file.

File: lib/mutant/cli.rb (hunk 2 above)
Comment (mbj) [inline lib/mutant/cli.rb]: You are correct. This code predates my preference for binary, non destructive operations by default. Even when the frozen values do not leak the context.

Will fix that.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -33,7 +33,7 @@
 require 'test_app'
 
 module Fixtures
-  TEST_CONFIG = Mutant::Config::DEFAULT.update(reporter: Mutant::Reporter::Trace.new)
+  TEST_CONFIG = Mutant::Config::DEFAULT.with(reporter: Mutant::Reporter::Trace.new)
```
Comment (dkubb) [inline spec/spec_helper.rb]: I know `#with` was added in Anima, but I wonder if maybe `#merge` would be a better alternative. It implies the object state is being merged with new state and that a new object is being returned. The `#with` method isn't commonly used for this purpose.

File: spec/support/corpus.rb [hunk 5]
```diff
@@ -134,6 +134,7 @@ def install_mutant
         repo_path.join('Gemfile').open('a') do |file|
           file << "gem 'mutant', path: '#{relative}'\n"
           file << "gem 'mutant-rspec', path: '#{relative}'\n"
+          file << "eval_gemfile File.expand_path('#{relative.join('Gemfile.shared')}')\n"
```
Comment (dkubb) [inline spec/support/corpus.rb]: If you decide to remove `Gemfile.shared` then this can be removed too.

File: spec/spec_helper.rb (hunk 4 above)
Comment (mbj) [inline spec/spec_helper.rb]: `#merge` implies a binary operation of two "equally shaped" objects. I especially thought about this one but than realized when we have an `anima` object on the left side, and a `Hash` instance on the right side: The operation is not commutative and should not be called `#merge`.

File: spec/support/corpus.rb (hunk 5 above)
Comment (mbj) [inline spec/support/corpus.rb]: Please see my comment about the intentions of `Gemfile.shared` here: https://github.com/mbj/mutant/pull/439#discussion_r38951885

**Comment by mbj** [comment]
@dkubb Ready for the next pass. All flags addressed (by code change or argument).

File: Gemfile (hunk 1 above)
Comment (dkubb) [inline Gemfile]: Ahh ok. I read the comment in the file, but the exact reason wasn't clear to me.

**Comment by dkubb** [comment]
@mbj I love that mutant self-mutates. By definition if a change doesn't result in the build breaking then it is ok to merge assuming the code passes review.

**Pull request merged**